lib/transmuxer/ts_transmuxer.js

```diff
@@ -0,0 +1,684 @@
+goog.require('shaka.util.Uint8ArrayUtils');
+
+
+/**
+ * @implements {shaka.extern.Transmuxer}
+ * @export
+ */
+shaka.transmuxer.TsTransmuxer = class {
+  /**
+   * @param {string} mimeType
+   */
+  constructor(mimeType) {
+    /** @private {string} */
+    this.originalMimeType_ = mimeType;
+
+    /** @private {number} */
+    this.frameIndex_ = 0;
+
+    /** @private {!Map.<number, !Uint8Array>} */
+    this.initSegments = new Map();
+
+    /** @private {?shaka.util.TsParser} */
+    this.tsParser_ = null;
+  }
+
+
+  /**
+   * @override
+   * @export
+   */
+  destroy() {
+    this.initSegments.clear();
+  }
+
+
+  /**
+   * Check if the mime type and the content type is supported.
+   * @param {string} mimeType
+   * @param {string=} contentType
+   * @return {boolean}
+   * @override
+   * @export
+   */
+  isSupported(mimeType, contentType) {
+    const Capabilities = shaka.media.Capabilities;
+
+    if (!this.isTsContainer_(mimeType)) {
+      return false;
+    }
+
+    const ContentType = shaka.util.ManifestParserUtils.ContentType;
+    const MimeUtils = shaka.util.MimeUtils;
+
+    const codecs = MimeUtils.getCodecs(mimeType);
+    const allCodecs = MimeUtils.splitCodecs(codecs);
+
+    const audioCodec = shaka.util.ManifestParserUtils.guessCodecsSafe(
+        ContentType.AUDIO, allCodecs);
+    const videoCodec = shaka.util.ManifestParserUtils.guessCodecsSafe(
+        ContentType.VIDEO, allCodecs);
+
+    // We don't support muxed content yet.
+    if (audioCodec && videoCodec) {
+      return false;
+    }
+
+    const TsTransmuxer = shaka.transmuxer.TsTransmuxer;
+
+    if (audioCodec) {
+      const normalizedCodec = MimeUtils.getNormalizedCodec(audioCodec);
+      if (!TsTransmuxer.SUPPORTED_AUDIO_CODECS_.includes(normalizedCodec)) {
+        return false;
+      }
+    }
+
+    if (videoCodec) {
+      const normalizedCodec = MimeUtils.getNormalizedCodec(videoCodec);
+      if (!TsTransmuxer.SUPPORTED_VIDEO_CODECS_.includes(normalizedCodec)) {
+        return false;
+      }
+    }
+
+    if (contentType) {
+      return Capabilities.isTypeSupported(
+          this.convertCodecs(contentType, mimeType));
+    }
+
+    const audioMime = this.convertCodecs(ContentType.AUDIO, mimeType);
+    const videoMime = this.convertCodecs(ContentType.VIDEO, mimeType);
+    return Capabilities.isTypeSupported(audioMime) ||
+        Capabilities.isTypeSupported(videoMime);
+  }
+
+
+  /**
+   * Check if the mimetype is 'video/mp2t'.
+   * @param {string} mimeType
+   * @return {boolean}
+   * @private
+   */
+  isTsContainer_(mimeType) {
+    return mimeType.toLowerCase().split(';')[0] == 'video/mp2t';
+  }
+
+
+  /**
+   * @override
+   * @export
+   */
+  convertCodecs(contentType, mimeType) {
+    if (this.isTsContainer_(mimeType)) {
+      const ContentType = shaka.util.ManifestParserUtils.ContentType;
+      const codecs = shaka.util.MimeUtils.getCodecs(mimeType)
+          .replace('mp4a.40.34', 'mp3');
```

Comment:
Isn't this part of getNormalizedCodec()?

Reply:
I think you are looking at another function, this is `convertCodecs` here we can't use `getNormalizedCodec` since we want to keep the original codec.
Firefox is the only browsers that supports mp3 in mp4. See:
`MediaSource.isTypeSupported('audio/mp4; codecs="mp4a.40.34"')` --> false
`MediaSource.isTypeSupported('audio/mp4; codecs="mp3"')` --> true

Reply:
A comment about that here would also be helpful.

Reply:
Added!